File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/xml/readxmlfile/vb/ReadXmlFile.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------

Imports Microsoft.VisualBasic
Imports System
Imports System.IO
Imports System.Xml

namespace HowTo.Samples.XML

public class ReadXmlFileSample

    private const document as String = "books.xml"

    Shared sub Main
        Dim myReadXmlFileSample as ReadXmlFileSample
        myReadXmlFileSample = new ReadXmlFileSample()
        myReadXmlFileSample.Run(document)
    end sub

    public sub Run(args as String)

        Dim reader as XmlTextReader = nothing

        try
            ' Load the file with an XmlTextReader
            Console.WriteLine ("Reading file books.xml ...")
            reader = new XmlTextReader (args)
            Console.WriteLine ("File books.xml read sucessfully ...")

            ' Process the supplied XML file
            Console.WriteLine ("Processing ...")
            Console.WriteLine ()
            FormatXml(reader, args)

        catch e as Exception

            Console.WriteLine ("Failed to read the file 0", args)
            Console.WriteLine ("Exception: {0}", e.ToString())

        finally
            Console.WriteLine()
            Console.WriteLine("Processing of the file {0} complete.", args)
            ' Finished with XmlTextReader
            If Not reader Is Nothing
                reader.Close()
            end if

        End try
    End Sub

    private shared Sub FormatXml (reader as XmlTextReader, filename as String)

        Dim declarationCount, piCount, docCount, commentCount, elementCount as Integer
        Dim attributeCount, textCount, whitespaceCount as Integer

        While reader.Read()

            Select (reader.NodeType)

            case XmlNodeType.XmlDeclaration:
                Format (reader, "XmlDeclaration")
                declarationCount += 1

            case XmlNodeType.ProcessingInstruction:
                Format (reader, "ProcessingInstruction")
                piCount += 1

            case XmlNodeType.DocumentType:
                Format (reader, "DocumentType")
                docCount += 1

            case XmlNodeType.Comment:
                Format (reader, "Comment")
                commentCount += 1

            case XmlNodeType.Element:
                Format (reader, "Element")
                elementCount += 1
                if (reader.HasAttributes)
                    attributeCount += reader.AttributeCount
                end if

            case XmlNodeType.Text:
                Format (reader, "Text")
                textCount += 1

            case XmlNodeType.Whitespace:
                whitespaceCount += 1

            End Select

        End While

        ' Display the Statistics for the file
        Console.WriteLine ()
        Console.WriteLine("Statistics for {0} file", filename)
        Console.WriteLine ()
        Console.WriteLine("XmlDeclaration: " & declarationCount)
        Console.WriteLine("ProcessingInstruction: " & piCount)
        Console.WriteLine("DocumentType: " & docCount)
        Console.WriteLine("Comment: " & commentCount)
        Console.WriteLine("Element: " & elementCount)
        Console.WriteLine("Element Attributes: " & attributeCount)
        Console.WriteLine("Text: " & textCount)
        Console.WriteLine("Whitespace: " & whitespaceCount)
    End Sub

    private shared Sub Format(byref reader as XmlTextReader , nodeType as String)

        ' Format the output
        Console.Write(reader.Depth & " ")
        Console.Write(reader.AttributeCount & " ")

        Dim i as Integer
        for i = 0 to reader.Depth - 1
            Console.Write(Strings.chr(9))
        Next

        Console.Write(nodeType & "<" & reader.Name & ">" & reader.Value)

        'Display the attributes values for the current node
        if (reader.HasAttributes)
            Console.Write(" Attributes:")
            Dim j as Integer
            for j = 0 to reader.AttributeCount - 1
                Console.Write(" [{0}] " & reader(j), j)
            next
        End if

        Console.WriteLine()

    End Sub

End Class
End Namespace

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/howto/samples/xml/readxmlfile/cp/ReadXmlFile.cpp
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------

#using <mscorlib.dll>
#using <System.Xml.Dll>

using namespace System;
using namespace System::IO;
using namespace System::Xml;

namespace HowToSamplesXML
{
__gc public class ReadXmlFileSample
{
protected:
    static String *document = "books.xml";

public:
    void Main()
    {
        ReadXmlFileSample* myReadXmlFileSample = new ReadXmlFileSample();
        myReadXmlFileSample->Run(document);
    }

    void Run(String* args)
    {
        XmlTextReader* reader = 0;

        try
        {
            // Load the file with an XmlTextReader
            Console::WriteLine (S"Reading file {0} ...", args);
            reader = new XmlTextReader (args);
            Console::WriteLine (S"File {0} read sucessfully ...", args);

            // Process the supplied XML file
            Console::WriteLine (S"Processing ...");
            Console::WriteLine ();
            FormatXml(reader, args);
        }
        catch (Exception* e)
        {
            Console::WriteLine (S"Failed to read the file {0}", args);
            Console::WriteLine (S"Exception: {0}", e->ToString());
        }

        __finally
        {
            Console::WriteLine();
            Console::WriteLine(S"Processing of the file {0} complete.", args);
            // Finished with XmlTextReader
            if (reader != 0)
                reader->Close();
        }
    }

private:
    void FormatXml (XmlReader* reader, String* filename)
    {
        int declarationCount=0, piCount=0, docCount=0, commentCount=0, elementCount=0, attributeCount=0, textCount=0, whitespaceCount=0;

        while (reader->Read())
        {
            switch (reader->NodeType)
            {
            case XmlNodeType::XmlDeclaration:
                Format (reader, S"XmlDeclaration");
                declarationCount++;
                break;
            case XmlNodeType::ProcessingInstruction:
                Format (reader, S"ProcessingInstruction");
                piCount++;
                break;
            case XmlNodeType::DocumentType:
                Format (reader, S"DocumentType");
                docCount++;
                break;
            case XmlNodeType::Comment:
                Format (reader, S"Comment");
                commentCount++;
                break;
            case XmlNodeType::Element:
                Format (reader, S"Element");
                while(reader->MoveToNextAttribute())
                {
                    Format (reader, S"Attribute");
                }
                elementCount++;
                if (reader->HasAttributes)
                    attributeCount += reader->AttributeCount;
                break;
            case XmlNodeType::Text:
                Format (reader, S"Text");
                textCount++;
                break;
            case XmlNodeType::Whitespace:
                whitespaceCount++;
                break;
            }
        }

        // Display the Statistics
        Console::WriteLine ();
        Console::WriteLine(S"Statistics for stream");
        Console::WriteLine();
        Console::WriteLine(S"XmlDeclaration: {0}",__box(declarationCount++));
        Console::WriteLine(S"ProcessingInstruction: {0}",__box(piCount++));
        Console::WriteLine(S"DocumentType: {0}",__box(docCount++));
        Console::WriteLine(S"Comment: {0}",__box(commentCount++));
        Console::WriteLine(S"Element: {0}",__box(elementCount++));
        Console::WriteLine(S"Attribute: {0}",__box(attributeCount++));
        Console::WriteLine(S"Text: {0}",__box(textCount++));
        Console::WriteLine(S"Whitespace: {0}",__box(whitespaceCount++));
    }

    // Format the output
    void Format(XmlReader* reader, String* NodeType)
    {
        // Format the output
        Console::Write(String::Concat(__box(reader->Depth), S" "));
        Console::Write(String::Concat(__box(reader->AttributeCount), S" "));
        for (int i=0; i < reader->Depth; i++)
        {
            Console::Write(S"\t");
        }

        Console::Write(String::Concat(NodeType, S"<", reader->Name, S">", reader->Value));
        // Display the attributes values for the current node
        if (reader->HasAttributes)
        {
            Console::Write(" Attributes:");

            for (int j=0; j < reader->AttributeCount; j++)
            {
                Console::Write(String::Concat(S" [" , __box(j) , S"] " , reader->Item[j]));
            }
        }

        Console::WriteLine();
    }

}; // End class ReadXmlFileSample
} // End namespace HowTo.Samples.XML

//Run the program
static void main()
{
    //Import the new package
    using namespace HowToSamplesXML;
    //Create the XPathExpressionSample class
    ReadXmlFileSample* myReadXmlFileSample = new ReadXmlFileSample();
    myReadXmlFileSample->Main();      

}




